### PR TITLE
use Travis-CI strict compile warnings/errors

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
This fixes a warning in libcperciva poll, and then enables strict Travis-CI compiler options.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 sudo: false
 dist: trusty
-compiler:
-  - clang
-  - gcc
-os:
-  - linux
-  - osx
+# We can't use strict flags in CFLAGS as a general environment
+# variable, because that messes up ./configure.
```
Comment (cperciva) [inline .travis.yml]: eh?

**Comment by cperciva** [comment]
Please split this into separate PRs; even with the common theme of "compiler warnings", adjusting network_select and .travis.yml in one merge is a bit disjointed.

File: .travis.yml (hunk 1 above)
Comment (gperciva) [inline .travis.yml]: err, right.  We need this for scrypt and tarsnap (eventually), but not other projects.

**Pull request merged**